n2survey/lime/survey.py

```diff
@@ -280,6 +315,75 @@ def get_responses(
 
         return responses
 
+    def __getitem__(
+        self, key: Union[pd.Series, pd.DataFrame, str, list, tuple]
+    ) -> Union["LimeSurvey", pd.DataFrame]:
+        """Retrieve or slice the responses DataFrame
+
+        Args:
+            key (pd.Series, pd.DataFrame, str, list, or tuple): A key for
+                DataFrame slicing or get_responses method
+
+        Returns:
+            LimeSurvey: A copy of LimeSurvey instance with filtered responses
+                DataFrame
+        """
+        filtered_survey = self.__copy__()
+
+        # A bool-valued Series, e.g. survey[survey.responses["A3"] == "A5"]
+        if isinstance(key, (pd.Series, pd.DataFrame)):
+            filtered_survey.responses = filtered_survey.responses[key]
+        # A question id, e.g. survey["A3"]
+        elif isinstance(key, str):
+            columns = filtered_survey.get_question(key).index
+            filtered_survey.responses = filtered_survey.responses[columns]
+        # A list of columns, e.g. survey[["C3_SQ001", "C3_Sq002"]]
+        elif isinstance(key, list):
+            columns = [
+                column
+                for question in key
+                for column in filtered_survey.get_question(question).index.to_list()
+            ]
+            filtered_survey.responses = filtered_survey.responses[columns]
+        # Two args, e.g. survey[survey.responses["A3"] == "A5", "B1"]
+        # or survey[1:10, ["B1", "C1_SQ001"]]
+        elif isinstance(key, tuple):
+            assert len(key) == 2
+            rows, columns = key
+            filtered_survey.responses = filtered_survey.responses[rows]
+            filtered_survey = filtered_survey[columns]
```

Comment:
What about `get_question`? Or if `columns` is a list of questions? I think it should work smoothly for those cases as well.
I think it would help if we restructure the code a little bit:
1. Define filters for rows (if there is any)
2. Define filters for columns (if there is any). If it is a string, convert it to a list
3. If there is filter for rows, filter rows
4. If there is filter for columns, filter columns

This way it will work the same for both tuple and non-tuple cases.